src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/CABI.java

```
@@ -32,9 +39,12 @@
 public enum CABI {
     SysV,
     Win64,
-    AArch64;
+    AArch64,
+    SysVppc64le,
+    SysVs390x,
+    AIX;
 
-    private static final CABI current;
+    private static CABI current;
```

This should remain `final`; throw an 'unsupported' exception for s390x if the OS is not Linux.

---

Updated the code as suggested above.